### ISSUE TITLE
[Bug] failed to create addons

### ISSUE THREAD

**Issue by matti** [weaveworks/eksctl#5438]
### What were you trying to accomplish?

create cluster with config:

```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: 'test-2'
  region: 'eu-north-1'
  version: '1.22'

addons:
  - name: vpc-cni
  - name: coredns
  - name: kube-proxy
iam:
  withOIDC: true


kubernetesNetworkConfig:
  ipFamily: IPv6
```

### What happened?

fails with:

```
2022-06-16 16:05:24 [ℹ]  creating addon
2022-06-16 16:05:35 [!]  1 error(s) occurred while creating addons
2022-06-16 16:05:35 [✖]  addon status transitioned to "DEGRADED"
Error: failed to create addons
```

**Logs**

```
2022-06-16 15:50:23 [ℹ]  eksctl version 0.100.0
2022-06-16 15:50:23 [ℹ]  using region eu-north-1
2022-06-16 15:50:23 [ℹ]  setting availability zones to [eu-north-1c eu-north-1a eu-north-1b]
2022-06-16 15:50:23 [ℹ]  subnets for eu-north-1c - public:192.168.0.0/19 private:192.168.96.0/19
2022-06-16 15:50:23 [ℹ]  subnets for eu-north-1a - public:192.168.32.0/19 private:192.168.128.0/19
2022-06-16 15:50:23 [ℹ]  subnets for eu-north-1b - public:192.168.64.0/19 private:192.168.160.0/19
2022-06-16 15:50:23 [ℹ]  using Kubernetes version 1.22
2022-06-16 15:50:23 [ℹ]  creating EKS cluster "test-2" in "eu-north-1" region with
2022-06-16 15:50:23 [ℹ]  will create a CloudFormation stack for cluster itself and 0 nodegroup stack(s)
2022-06-16 15:50:23 [ℹ]  will create a CloudFormation stack for cluster itself and 0 managed nodegroup stack(s)
2022-06-16 15:50:23 [ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=eu-north-1 --cluster=test-2'
2022-06-16 15:50:23 [ℹ]  Kubernetes API endpoint access will use default of {publicAccess=true, privateAccess=false} for cluster "test-2" in "eu-north-1"
2022-06-16 15:50:23 [ℹ]  CloudWatch logging will not be enabled for cluster "test-2" in "eu-north-1"
2022-06-16 15:50:23 [ℹ]  you can enable it with 'eksctl utils update-cluster-logging --enable-types={SPECIFY-YOUR-LOG-TYPES-HERE (e.g. all)} --region=eu-north-1 --cluster=test-2'
2022-06-16 15:50:23 [ℹ]
2 sequential tasks: { create cluster control plane "test-2",
    5 sequential sub-tasks: {
        wait for control plane to become ready,
        associate IAM OIDC provider,
        no tasks,
        restart daemonset "kube-system/aws-node",
        1 task: { create addons },
    }
}
2022-06-16 15:50:23 [ℹ]  building cluster stack "eksctl-test-2-cluster"
2022-06-16 15:50:24 [ℹ]  deploying stack "eksctl-test-2-cluster"
2022-06-16 15:50:54 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 15:51:24 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 15:52:24 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 15:53:25 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 15:54:25 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 15:55:25 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 15:56:25 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 15:57:26 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 15:58:26 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 15:59:26 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 16:00:26 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 16:01:27 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-cluster"
2022-06-16 16:03:30 [ℹ]  daemonset "kube-system/aws-node" restarted
2022-06-16 16:03:30 [ℹ]  creating role using recommended policies
2022-06-16 16:03:31 [ℹ]  deploying stack "eksctl-test-2-addon-vpc-cni"
2022-06-16 16:03:31 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-addon-vpc-cni"
2022-06-16 16:04:01 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-addon-vpc-cni"
2022-06-16 16:04:32 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-addon-vpc-cni"
2022-06-16 16:05:06 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-addon-vpc-cni"
2022-06-16 16:05:06 [ℹ]  creating addon
2022-06-16 16:05:17 [ℹ]  addon "vpc-cni" active
2022-06-16 16:05:17 [ℹ]  waiting for the control plane availability...
2022-06-16 16:05:23 [✔]  saved kubeconfig as "/root/.kube/aws-eks-eu-north-1-test-2"
2022-06-16 16:05:23 [ℹ]  no tasks
2022-06-16 16:05:23 [✔]  all EKS cluster resources for "test-2" have been created
2022-06-16 16:05:24 [ℹ]  no recommended policies found, proceeding without any IAM
2022-06-16 16:05:24 [ℹ]  creating addon
2022-06-16 16:05:35 [!]  1 error(s) occurred while creating addons
2022-06-16 16:05:35 [✖]  addon status transitioned to "DEGRADED"
Error: failed to create addons
        "Arn": "arn:aws:eks:eu-north-1:asdf:cluster/test-2",
        "CertificateAuthority": {
            "Data": "removed"
        },
        "ClientRequestToken": null,
        "ConnectorConfig": null,
        "CreatedAt": "2022-06-16T15:51:34.373Z",
        "EncryptionConfig": null,
        "Endpoint": "https://asdf.sk1.eu-north-1.eks.amazonaws.com",
        "Identity": {
            "Oidc": {
                "Issuer": "https://oidc.eks.eu-north-1.amazonaws.com/id/asdf"
            }
        },
        "KubernetesNetworkConfig": {
            "IpFamily": "ipv6",
            "ServiceIpv4Cidr": null,
            "ServiceIpv6Cidr": "fdcf:0603:8aae::/108"
        },
        "Logging": {
            "ClusterLogging": [
                {
                    "Enabled": false,
                    "Types": [
                        "api",
                        "audit",
                        "authenticator",
                        "controllerManager",
                        "scheduler"
                    ]
                }
            ]
        },
        "Name": "test-2",
        "PlatformVersion": "eks.2",
        "ResourcesVpcConfig": {
            "ClusterSecurityGroupId": "sg-058432f560d3f156e",
            "EndpointPrivateAccess": false,
            "EndpointPublicAccess": true,
            "PublicAccessCidrs": [
                "0.0.0.0/0"
            ],
            "SecurityGroupIds": [
                "sg-03022cb62c114ed3f"
            ],
            "SubnetIds": [
                "subnet-0c31083bc8246a043",
                "subnet-004004bf789b348ae",
                "subnet-0ceaebdfb98686d54",
                "subnet-0821697ee16452674",
                "subnet-0e274b818b3b1c156",
                "subnet-03428e90598516b5d"
            ],
            "VpcId": "vpc-020f50ff06102fee8"
        },
        "RoleArn": "arn:aws:iam::asdf:role/eksctl-test-2-cluster-ServiceRole-KD7W4QYG1KBZ",
        "Status": "ACTIVE",
        "Tags": {
            "Name": "eksctl-test-2-cluster/ControlPlane",
            "alpha.eksctl.io/cluster-name": "test-2",
            "alpha.eksctl.io/eksctl-version": "0.100.0",
            "aws:cloudformation:logical-id": "ControlPlane",
            "aws:cloudformation:stack-id": "arn:aws:cloudformation:eu-north-1:asdf:stack/eksctl-test-2-cluster/08e06210-ed8c-11ec-9bc3-067360a02c5a",
            "aws:cloudformation:stack-name": "eksctl-test-2-cluster",
            "eksctl.cluster.k8s.io/v1alpha1/cluster-name": "test-2"
        },
        "Version": "1.22"
    }
]
2022-06-16 16:05:48 [ℹ]  deleting EKS cluster "test-2"
```

**Versions**
```
$ eksctl info
eksctl version: 0.100.0
kubectl version: v1.23.4
OS: linux
```

**Comment by matti** [comment]
is it so that `addons:` requires `version: latest` for each? I think that's the only change I have made

```
addons:
  - name: vpc-cni
  - name: coredns
  - name: kube-proxy
```

**Comment by Skarlso** [comment]
Nope, that's coredns failing because you don't have any nodes. Once you have nodes, it won't fail.

**Comment by matti** [comment]
but... I have previously created a cluster without any nodes and it worked (exit 0) just fine.

Has this now changed so that I need to first create a cluster without any addons, then add nodes and then add addons?

**Comment by Skarlso** [comment]
You mean you previously created an ipv6 cluster without nodes? Can you check why it's failing though? Maybe something changed with the addon that we now have to take into account. This is not eksctl failing.

**Comment by matti** [comment]
yep, I have created about 200 clusters with eksctl without nodes and it has worked so far.

how do I check whats the failure?

my plan is now remove addons from cluster yaml and add them after nodes, its a better flow like that anyway

**Comment by Skarlso** [comment]
>yep, I have created about 200 clusters with eksctl without nodes and it has worked so far.

Were they ipv6?

**Comment by Skarlso** [comment]
Note, this failed for a normal, console created EKS cluster as well. So I don't think this is something eksctl is doing.

**Comment by Skarlso** [comment]
Ah this is actually this issue https://github.com/weaveworks/eksctl/issues/4030

**Comment by Himangini** [comment]
I believe this issue is fixed by #4030, https://github.com/weaveworks/eksctl/pull/5458. 

@matti are you still having issues?

**Comment by matti** [comment]
@Himangini still broken https://github.com/weaveworks/eksctl/issues/5570
